src/prefect/agent/agent.py

```diff
@@ -593,29 +613,34 @@ def _get_flow_run_metadata(
         flow_run_ids = list(flow_run_ids)
         self.logger.debug(f"Retrieving metadata for {len(flow_run_ids)} flow run(s)...")
 
+        # This buffer allows flow runs to retry immediately in their own deployment
+        # without the agent creating a second deployment
+        retry_start_time_buffer = pendulum.now("UTC").subtract(seconds=3).isoformat()
+
+        where = {
```

Comment:
should we also be doing filtering based on agent labels here?

Reply:
`get_runs_in_queue` takes care of that already